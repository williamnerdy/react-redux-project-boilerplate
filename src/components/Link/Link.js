import React from 'react';
import { Link as RouterLink } from 'react-router-dom';

const Link = ({ href, to, children }) => {
  const props = {
    className: 'link'
  };

  return href ? (
    <a href={href} {...props}>
      {children}
    </a>
  ) : (
    <RouterLink to={to} {...props}>
      {children}
    </RouterLink>
  );
};

export default Link;
