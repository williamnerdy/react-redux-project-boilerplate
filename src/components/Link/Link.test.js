import React from 'react';
import { shallow } from 'enzyme';
import Link from './Link';

describe('<Link />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Link href="#">My link</Link>);
  });

  it('Renders without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
