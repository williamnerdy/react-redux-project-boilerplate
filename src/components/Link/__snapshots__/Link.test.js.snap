// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> Renders without crashing 1`] = `
<a
  className="link"
  href="#"
>
  My link
</a>
`;
